frontend/express/public/javascripts/countly/vue/components/dropdown.js

```diff
@@ -187,6 +187,123 @@
+                                :disabled="false"\
+                                :adaptive-length="false"\
+                                :focused="dropdown.focused"\
+                                :opened="dropdown.visible"\
+                                :placeholder="label"\>\
+                            </cly-input-dropdown-trigger>\
+                        </slot>\
+                        </template>\
+                        <div class="cly-vue-menubox-new menubox-default-skin">\
+                        <div class="menu-body">\
+                            <div>\
+                            <div>\
+                            <span class="cly-vue-menubox-new__title">{{title}}</span>\
+                            <span class="cly-vue-menubox-new__reset" @click="reset"> Reset Filters</span>\
+                            </div>\
+                            <table  \
+                                v-for="field in fields" :key="field.key">\
+					            <tr class="cly-vue-menubox-new__field">{{ field.label }}</tr>\
+					            <tr>\
+						            <el-select class="cly-vue-menubox-new__field-dropdown" :placeholder="internalValue[field.key].name?internalValue[field.key].name:internalValue[field.key]" v-model="internalValue[field.key]">\
+							            <el-option v-for="item in field.items" :key="item.key"\
+								        :value="item.name">\
+							            </el-option>\
+						            </el-select>\
+					            </tr>\
+			            	</table>\
+                            </div>\
+                            <div class="controls">\
+                            <el-button v-bind="$attrs" class="cly-vue-menubox-new__cancel" @click="close">  {{cancelLabel}}\
+                            </el-button>\
+                            <el-button v-bind="$attrs" class="cly-vue-menubox-new__confirm" @click="save">  {{confirmLabel}}\
+                            </el-button>\
+                            </div>\
+                        </div>\
+                        </div>\
+                    </cly-dropdown>',
+        props: {
+            cancelLabel: {
+                type: String,
+                default: 'Cancel'
+            },
+            confirmLabel: {
+                type: String,
+                default: 'Confirm'
+            },
+            value: {
+                type: Object,
+                default: function() {
+                    return {};
+                }
+            },
+            fields: {
+                type: Array,
+                default: function() {
+                    return [];
+                }
+            },
+            title: {
+                type: String,
+                default: ''
+            }
+
+        },
+        data: function() {
+            return {
+                internalValue: function() {
+                    return {};
+                },
+            };
+        },
+        watch: {
+            value: {
```

Comment:
do we need to watch it?
i believe value never changes since its not being used in the template.
i speculate its only changed if the parent changes it.

Reply:
I think this is fine Prikshit. This mechanism ensures that parent changes are always reflected on the component, while the opposite is not, so that it requires the user's explicit confirmation for value update.

Reply:
ok got it.

Reply:
but even if internalValue is changed within the component, it wont reflect in the parent bcz we are not changing value.
I suppose if we set internalValue as computed property it should work.

Reply:
So basically without the watcher here , using the computedProperty the value will change while selecting the internal dropdowns and hence that would effect the main dropdown label :)

Reply:
@itsiprikshit But we still want to modify internalValue. Setting it computed would make it read-only, right?

Reply:
Hey @can-keklik . I think @itsiprikshit meant something like this...        internalValue: function() {
                return this.value;
            }